upi/aws/cloudformation/02_cluster_infra.yaml

```diff
@@ -7,6 +7,9 @@ Parameters:
     ConstraintDescription: Cluster name must be alphanumeric, start with a letter and a maximum of 32 characters
     Description: A short, representative cluster name to use for hostnames, etc.
     Type: String
+  InfrastructureID:
```

Comment:
`InfrastructureID ` appears to be used only as an input to functions which compute the cluster ID, e.g. `{clusterName}-{infraID}`.

The `jq` command above already provides the cluster ID.

Would another possible option be to instead introduce a`ClusterID` parameter which is used directly in `HostedZoneTags`?

Reply:
You are correct. This is possible, but we also lose flexibility. In CloudFormation, there is no split( ) function, so we'd be stuck with the combined string inside the template. By passing in the parts, we have more flexibility later without having to change the API (parameters)

Reply:
Is cluster ID supposed to be opaque? By leaking details of the algorithm here, is any risk introduced with regards to how we compute the value in the future? (Another way: what happens if we later change the way cluster ID is computed?)

Reply:
That. I do not know. @wking @crawford @abhinavdahiya suggests?

Reply:
I think we want the cluster name and infra ID as separate, opaque strings.  We should never need the random suffix without the cluster-name prefix, and, as discussed above, separate strings isolates us from algorithm changes (although those seem unlikely as well).

Reply:
Can we switch to having `InfrastructureID` be the whole string (`openshift-vw9j6`) and not just the uniquifying suffix, for the DRYness reasons discussed above?  I am willing to eat another round of CI tests to get this change into this PR ;).